src/packages/options/src/create.ts

```diff
@@ -38,22 +36,22 @@ const checkForConflicts = (
   }
 };
 
-function fill(defaults: any, options: any, target: any, namespace: any) {
-  const def = defaults[namespace];
+function fill(options: any, target: any, def: any, namespace: any) {
   const config = (target[namespace] = target[namespace] || {});
   const flavor = options.flavor;
 
   const suppliedOptions = new Set<string>();
-  const keys = Object.keys(def);
+  const entries = Object.entries(def) as [string, any][];
   if (hasOwn(options, namespace)) {
     const namespaceOptions = options[namespace];
 
-    for (let i = 0, l = keys.length; i < l; i++) {
-      const key = keys[i];
-      const propDefinition = def[key];
+    for (let i = 0, l = entries.length; i < l; i++) {
+      const entry = entries[i];
+      const key = entry[0];
+      const propDefinition = entry[1];
```

Comment:
```suggestion
    for (const [key, propDefinition] of entries) {
```

Reply:
https://github.com/LeanyLabs/js-array-performance tells me that `for of` is the fastest array iteration technique, and it's super tidy, especially with the array destructuring.